### PR TITLE
Improve `wrangler dev` error message for invalid tokens.

### PR DESCRIPTION
**Pull request by ocsfrank**
This should display more info when dev fails(issue #2054) because of invalid tokens or invalid request headers.

### CODE REVIEW

File: src/commands/whoami.rs [hunk 1]
```diff
@@ -116,7 +116,24 @@ pub(crate) fn fetch_accounts(user: &GlobalUser) -> Result<Vec<Account>> {
...
     match response {
         Ok(res) => Ok(res.result),
-        Err(e) => anyhow::bail!(http::format_error(e, None)),
+        Err(e) => {
+            match e {
+                ApiFailure::Error(_, ref api_errors) => {
+                    let error = &api_errors.errors[0];
+                    if error.code == 9109 {
+                        // 9109 error code = Invalid access token
+                        StdOut::info("Token might be expired or might not have the necessary permissions. Please consider running `wranger logout` or re-authenticate wrangler through `wrangler login` or `wrangler config`.");
```
Comment (threepointone) [inline src/commands/whoami.rs:125]: This will require https://github.com/cloudflare/wrangler/pull/2048 to land so wrangler logout can work, yes?

Comment (threepointone) [inline src/commands/whoami.rs:125]:
```suggestion
                        StdOut::info("Your API token might be expired, or might not have the necessary permissions. Please re-authenticate wrangler by running `wrangler login` or `wrangler config`.");
```
Maybe we shouldn't even suggest logging out, if wrangler login will wipe out the previous token?

File: src/commands/whoami.rs [hunk 2]
```diff
@@ -116,7 +116,24 @@ pub(crate) fn fetch_accounts(user: &GlobalUser) -> Result<Vec<Account>> {
...
+                    } else if error.code == 6003 {
+                        // 6003 error code = Invalid request headers. A common case is when the value of an authorization method has been changed outside of wrangler commands
+                        StdOut::info("Your authentication method might be corrupted (e.g. API token value has been altered). Please consider re-authenticating wrangler through `wrangler login` or `wrangler config`.")
```
Comment (threepointone) [inline src/commands/whoami.rs:128]: Similar suggestion here. Maybe we shouldn't recommend logging out, when they'll have to do a login again anyway (and if login wipes out the previous token)

**Review by threepointone — APPROVED** [review]
Approving with suggestions.

File: src/commands/whoami.rs (hunk 1 above)
Comment (ocsfrank) [reply src/commands/whoami.rs:125]: Oops. Yes, wrangler login will completely override the previous token.

**Pull request merged**